sdks/java/core/src/main/java/org/apache/beam/sdk/testing/PAssert.java

```diff
@@ -127,12 +129,58 @@ private PAssert() {}
      * Creates a new {@link IterableAssert} like this one, but with the assertion restricted to only
      * run on the provided window.
      *
+     * <p>The assertion will expect outputs to be produced to the provided window exactly once. If
+     * the upstream {@link Trigger} may produce output multiple times, consider instead using
+     * {@link #inFinalPane(BoundedWindow)}, {@link #inOnTimePane(BoundedWindow)} or
+     * {@link #acrossAllPanes(BoundedWindow)}.
+     *
      * @return a new {@link IterableAssert} like this one but with the assertion only applied to the
      * specified window.
      */
     IterableAssert<T> inWindow(BoundedWindow window);
 
     /**
+     * Creates a new {@link IterableAssert} like this one, but with the assertion restricted to only
+     * run on the provided window, running the checker only on the final pane for each key.
+     *
+     * <p>If the input {@link WindowingStrategy} does not always produce final panes (with
+     * {@link ClosingBehavior#FIRE_ALWAYS}), this may produce an empty output even if the trigger
+     * has fired previously.
+     *
+     * @return a new {@link IterableAssert} like this one but with the assertion only applied to the
+     * specified window.
+     */
+    IterableAssert<T> inFinalPane(BoundedWindow window);
+
+    /**
+     * Creates a new {@link IterableAssert} like this one, but with the assertion restricted to only
+     * run on the provided window.
+     *
+     * @return a new {@link IterableAssert} like this one but with the assertion only applied to the
+     * specified window.
+     */
+    IterableAssert<T> inOnTimePane(BoundedWindow window);
+
+    /**
+     * Creates a new {@link IterableAssert} like this one, but with the assertion restricted to only
+     * run on the provided window across all panes that were not produced by the arrival of late
+     * data.
+     *
+     * @return a new {@link IterableAssert} like this one but with the assertion only applied to the
+     * specified window.
+     */
+    IterableAssert<T> acrossNonLatePanes(BoundedWindow window);
```

Comment:
Perhaps `combinedNonLatePanes`? My understanding of the use of "across" was to avoid the ambiguity that might arise if you said `inNonLatePanes`, since that could mean in all of them combined or in all of them individually.